src/MSRIOGroup.cpp

```diff
@@ -285,6 +288,82 @@ namespace geopm
         }
     }
 
+    void MSRIOGroup::register_rdt_signals(void)
+    {
+        // It may be necessary to wrap the initial rdt() call in this check
+        int domain = signal_domain_type("MSR::MISC_ENABLE:LIMIT_CPUID_MAXVAL");
+        int num_domain = m_platform_topo.num_domain(domain);
+        bool disable = false;
+        for (int dom_idx = 0; dom_idx < num_domain; ++dom_idx) {
+            disable |= (bool)read_signal("MSR::MISC_ENABLE:LIMIT_CPUID_MAXVAL", domain, dom_idx);
+        }
+        if (disable || (!m_rdt.rdt_support)) {
+            return;
+        }
+
+        std::string signal_name = "QM_CTR_SCALED";
+        std::string msr_name = "MSR::QM_CTR:RM_DATA";
+        std::string description = "Resource Monitor Data converted to bytes";
+
+        auto read_it = m_signal_available.find(msr_name);
+        if (read_it != m_signal_available.end()) {
+            auto readings = read_it->second.signals;
+            int ctr_domain = read_it->second.domain;
+
+            // Begin bit, end bit, and function are currently hardcoded as there's no accessor
+            // for MSRFieldSignal values that doesn't involve reparsing the JSON file
+            MSRIOGroup::add_msr_field_signal("QM_CTR",
+                                             signal_name,
+                                             ctr_domain,
+                                             0, 31,
```

Comment:
We can't easily get begin and end bit for a signal once it's created.  Hard-coding range here because of that.

Reply:
Okay

Reply:
Update MSRIOGroup to provide a new signal exposing RMID scalar (MSR::RMID_QM_SCALAR?), and a combination signal that multiplies RMID Scalar Value and QM_CTR raw.